feat: platform config footprint file parser

ArnavK-09 opened the pull request:
related 

https://github.com/tscircuit/tscircuit/issues/768

prolly cos of props?


<img width="1201" height="46" alt="image" src="https://github.com/user-attachments/assets/c78bed7a-8968-4ff5-9acc-9bee4b80ed81" />

<img width="1193" height="37" alt="image" src="https://github.com/user-attachments/assets/4c936088-a49a-49d6-a7bc-47e306cb22fd" />

ArnavK-09 committed:
feat: platform config footprint file parser

ArnavK-09 committed:
format

seveibar commented on tests/examples/example20-kicad-import.test.tsx:
```diff
@@ -0,0 +1,213 @@
+        ...basePlatformConfig,
+        footprintFileParserMap: {
+          kicad_mod: {
+            loadFromUrl: async (url: string) => {
```
good, but i think you want to make this a part of the default platform config right? getDefaultPlatformConfig or something

seveibar reviewed (approved):
a good step, but i think you want to add it to the default

ArnavK-09 committed:
Add Kicad footprint parser to platform config

Implement a parser for Kicad footprint files in the platform
configuration, allowing for direct import of Kicad components into the
circuit framework.

ArnavK-09 committed:
testing

ArnavK-09 committed:
Discard changes to .github/workflows/bun-test.yml
